File: JS/BOJ-B2-1159-Basketball_game.js
"use strict";

const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
let input = fs
  .readFileSync(filePath)
  .toString()
  .trim()
  .split("\n")
  .map((v) => v.trimEnd());

let output = [];
const solution = (input) => {
  const n = input[0];
  const names = input.slice(1);
  const arr = Array.from({length: 26}, () => 0);

  for(let i = 0; i < n; ++i) {
    arr[names[i].charCodeAt(0) - 97]++;
  }

  for(let i = 0; i < arr.length; ++i) {
    if(arr[i] >= 5) output.push(String.fromCharCode(i + 97));
  }

  if(!output.length) output.push('PREDAJA');
};

solution(input);
// fs.writeFileSync("/dev/stdout", output.join(""));
fs.writeFileSync("../output.txt", output.join(""), (err) => console.error);

File: CPP/BOJ-B2-1159-Basketball_game.cpp
#include <bits/stdc++.h>
using namespace std;
int n, cnt[26];
string str, ret;

int main()
{
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);
  // freopen("input.txt", "r", stdin);
  // freopen("output.txt", "w", stdout);

  cin >> n;
  for (int i = 0; i < n; ++i)
  {
    cin >> str;
    // 정확한 형태는 int(str[0] - 'a')
    cnt[str[0] - 'a']++;
  }
  for (int i = 0; i < 26; ++i)
  {
    if (cnt[i] >= 5)
      // ret은 string 선언이기 때문에 숫자로 들어가도 문자로 저장된다.
      // 정확한 형태는 char(i + 'a')
      ret += (i + 'a');
  }
  if (ret.size())
    cout << ret << "\n";
  else
    cout << "PREDAJA" << "\n";

  return 0;
}